api/document/js/fixup-doc.ts

```diff
@@ -0,0 +1,40 @@
+import { Node } from 'prosemirror-model';
+import { EditorState, Plugin, Transaction } from 'prosemirror-state';
+
+import schema from './schema';
+
+const shouldDeleteChecks = {
+  list: (node: Node) => node.content.childCount === 0,
+  listitem: (node: Node) => node.content.childCount === 0,
+  para: (node: Node) => (
```

Comment:
I wish these sorts of things could somehow be embedded as methods (alongside e.g. `toDOM()`) on the nodes' schema entries.  It'd provide a bit more context for why the schemas _seem_ to allow for empty content, and it'd also make it harder to forget to add "should delete" logic related to a node... one concern with putting this object here is that folks who add new nodes to the schema might forget to think about whether they should also make an entry for the new node here, too.

Reply:
What do you think about creating a wrapper around the ProseMirror Node schema to collect all of these types of methods? Then we'd ensure we had factories, serializers, etc. in one place.